### ISSUE TITLE
Include `for of` loops as well

### ISSUE THREAD

**Issue by wbobeirne** [buildo/eslint-plugin-no-loops#3]
These rules have been great for keeping our team from needing a `Symbol` shim by avoiding using for loops, and instead using array functions. The problem is that `for (iterator of list)` is still slipping through since there are no rules for it. I tried forking the plugin and doing it myself, but I'm having some trouble getting ESLint to like ES6 rules, the test I wrote for it fails with a `Parsing error: Unexpected token of`.

If this can get added, it would be much appreciated!

**Comment by gabro** [comment]
It should as easy as

``` diff
  return {
    ForStatement: reportLoopPresence,
    ForInStatement: reportLoopPresence,
    WhileStatement: reportLoopPresence,
-   DoWhileStatement: reportLoopPresence
+   DoWhileStatement: reportLoopPresence,
+   ForOfStatement: reportLoopPresence
  };
```

Did you try something like this? Anyway, yes, I think it's a valuable addition

**Comment by wbobeirne** [comment]
Yeah, that was the easy part. But I wanted to add a test to make sure it was super kosh, and that was the part where ES6 was giving me guff. I'll leave that up to you to decide how to deal with that, or if it needs a  test.

**Comment by gabro** [comment]
oh, I see, then yes, let me try it. It should be just a matter of configuring the parser

**Comment by gabro** [comment]
@wbobeirne ok, done. I had to use babel-eslint in the tests ;)

**Comment by gabro** [comment]
released as v0.3.0, enjoy 👍
